app/grandchallenge/evaluation/models.py

```diff
@@ -1172,11 +1192,80 @@ class SubmissionGroupObjectPermission(GroupObjectPermissionBase):
     content_object = models.ForeignKey(Submission, on_delete=models.CASCADE)
 
 
+def ground_truth_path(instance, filename):
+    return (
+        f"ground_truths/"
+        f"{instance._meta.app_label.lower()}/"
+        f"{instance._meta.model_name.lower()}/"
+        f"{instance.pk}/"
+        f"{get_valid_filename(filename)}"
+    )
+
+
+class EvaluationGroundTruth(Tarball):
+    phase = models.ForeignKey(
+        Phase, on_delete=models.PROTECT, related_name="ground_truths"
+    )
+    ground_truth = models.FileField(
+        blank=True,
+        upload_to=ground_truth_path,
+        validators=[ExtensionValidator(allowed_extensions=(".tar.gz",))],
+        help_text=(
+            ".tar.gz file of the ground truth that will be extracted to /opt/ml/input/data/ground_truth/ during inference"
+        ),
+        storage=protected_s3_storage,
+    )
+
+    def assign_permissions(self):
+        # Challenge admins can view this ground truth
+        assign_perm(
+            f"view_{self._meta.model_name}",
+            self.phase.challenge.admins_group,
+            self,
+        )
+        # Challenge admins can change this ground truth
+        assign_perm(
+            f"change_{self._meta.model_name}",
+            self.phase.challenge.admins_group,
+            self,
+        )
+
+    def get_peer_tarballs(self):
+        return EvaluationGroundTruth.objects.filter(phase=self.phase).exclude(
+            pk=self.pk
+        )
+
+    def get_absolute_url(self):
+        return reverse(
+            "evaluation:ground-truth-detail",
+            kwargs={
+                "slug": self.phase.slug,
+                "pk": self.pk,
+                "challenge_short_name": self.phase.challenge.short_name,
+            },
+        )
+
+
+class EvaluationGroundTruthUserObjectPermission(UserObjectPermissionBase):
+    content_object = models.ForeignKey(
+        EvaluationGroundTruth, on_delete=models.CASCADE
+    )
+
+
+class EvaluationGroundTruthGroupObjectPermission(GroupObjectPermissionBase):
+    content_object = models.ForeignKey(
+        EvaluationGroundTruth, on_delete=models.CASCADE
+    )
+
+
 class Evaluation(UUIDModel, ComponentJob):
     """Stores information about a evaluation for a given submission."""
 
     submission = models.ForeignKey("Submission", on_delete=models.PROTECT)
     method = models.ForeignKey("Method", on_delete=models.PROTECT)
+    ground_truth = models.ForeignKey(
+        EvaluationGroundTruth, null=True, blank=True, on_delete=models.SET_NULL
```

Comment:
Probably this should be protected, rather than set to null. Same should go for the algorithm job - model relationship. We would otherwise lose the information about what the job/evaluation was run with.

Reply:
True. On the algorithm job model this already the case, but not on the submission model. Sorry, I copied how the algorithm image is added on each of those models and didn't realize the discrepancy. Should we also set the on_delete for the algorithm_image on the submission to Protect?

Reply:
Done